packages/datetime/test/dateRangeInputTests.tsx

```diff
@@ -122,12 +137,247 @@ describe("<DateRangeInput>", () => {
+                    input.simulate("blur");
+                    assertInputTextEquals(input, OUT_OF_RANGE_MESSAGE);
+                };
+                _runTestForEachScenario(_runTest);
+            });
+
+            describe("shows the offending date in the field on focus", () => {
+                const _runTest = (input: WrappedComponentInput, inputString: string) => {
+                    changeInputText(input, inputString);
+                    input.simulate("blur");
+                    input.simulate("focus");
+                    assertInputTextEquals(input, inputString);
+                };
+                _runTestForEachScenario(_runTest);
+            });
+
+            describe("calls onError with invalid date on blur", () => {
+                const _runTest = (input: WrappedComponentInput, inputString: string, boundary: DateRangeBoundary) => {
+                    const expectedRange = (boundary === DateRangeBoundary.START)
+                        ? [inputString, null]
+                        : [null, inputString];
+                    input.simulate("focus");
+                    changeInputText(input, inputString);
+                    expect(onError.called).to.be.false;
+                    input.simulate("blur");
+                    expect(onError.calledOnce).to.be.true;
+                    assertDateRangesEqual(onError.getCall(0).args[0], expectedRange);
+                };
+                _runTestForEachScenario(_runTest);
+            });
+
+            describe("does NOT call onChange before OR after blur", () => {
+                const _runTest = (input: WrappedComponentInput, inputString: string) => {
+                    input.simulate("focus");
+                    changeInputText(input, inputString);
+                    expect(onChange.called).to.be.false;
+                    input.simulate("blur");
+                    expect(onChange.called).to.be.false;
+                };
+                _runTestForEachScenario(_runTest);
+            });
+
+            describe("removes error message if input is changed to an in-range date again", () => {
+                const _runTest = (input: WrappedComponentInput, inputString: string) => {
+                    changeInputText(input, inputString);
+                    input.simulate("blur");
+
+                    const IN_RANGE_DATE_STR = START_STR;
+                    input.simulate("focus");
+                    changeInputText(input, IN_RANGE_DATE_STR);
+                    input.simulate("blur");
+                    assertInputTextEquals(input, IN_RANGE_DATE_STR);
+                };
+                _runTestForEachScenario(_runTest);
+            });
+
+            type OutOfRangeTestFunction = (input: WrappedComponentInput,
+                                           inputString: string,
+                                           boundary?: DateRangeBoundary) => void;
+
+            function _runTestForEachScenario(runTestFn: OutOfRangeTestFunction) {
```

Comment:
no need for _ prefix here because it's not exported so it's like _super_ private--literally can't even be accessed from outside this `describe` block.